File: libraries/sorting-library-csharp-test/SortingProcessorTest.cs
using sorting_library_csharp;

namespace sorting_library_csharp_test;

public class SortingProcessorTest
{
    private List<int> sampleArray;

    private static bool IsSorted(List<int> array)
    {
        for (int i = 0; i < array.Count - 1; i++)
        {
            if (array[i] > array[i + 1])
            {
                return false;
            }
        }
        return true;
    }

    public SortingProcessorTest()
    {
        sampleArray = new List<int> { 5, 3, 4, 1, 2 };
    }

    [Fact]
    public void SortInON1()
    {
        // Act
        SortingProcessor.SortInON1(sampleArray);
        // Assert
        Assert.True(IsSorted(sampleArray), "sample array is sorted");
    }

    [Fact]
    public void SortInON2()
    {
        // Act
        SortingProcessor.SortInON2(sampleArray);
        // Assert
        Assert.True(IsSorted(sampleArray), "sample array is sorted");
    }

    [Fact]
    public void SortInON3()
    {
        // Act
        SortingProcessor.SortInON3(sampleArray);
        // Assert
        Assert.True(IsSorted(sampleArray), "sample array is sorted");
    }
}

File: standalone-apps/sorting-api-spring/src/test/java/org/experimental/app/service/SortingProcessorTest.java
package org.experimental.app.service;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertTrue;

class SortingProcessorTest {

    private ArrayList<Integer> sampleArray;

    private static boolean isSorted(List<Integer> array) {
        for (int i = 0; i < array.size() - 1; i++) {
            if (array.get(i) > array.get(i + 1)) {
                return false;
            }
        }
        return true;
    }

    @BeforeEach
    void prepareArray() {
        sampleArray = new ArrayList<>(List.of(5, 3, 4, 1, 2));
    }

    @Test
    void sortInON1() {
        // Act
        SortingProcessor.sortInON1(sampleArray);
        // Verify
        assertTrue(isSorted(sampleArray), "sample array is sorted");
    }

    @Test
    void sortInON2() {
        // Act
        SortingProcessor.sortInON2(sampleArray);
        // Verify
        assertTrue(isSorted(sampleArray), "sample array is sorted");
    }

    @Test
    void sortInON3() {
        // Act
        SortingProcessor.sortInON3(sampleArray);
        // Verify
        assertTrue(isSorted(sampleArray), "sample array is sorted");
    }
}
